ci-operator/step-registry/cucushift/installer/check/ibmcloud/resourcegroup/cucushift-installer-check-ibmcloud-resourcegroup-commands.sh

```diff
@@ -0,0 +1,107 @@
+#!/bin/bash
+
+set -o nounset
+set -o errexit
+set -o pipefail
+
+function ibmcloud_login {
+    export IBMCLOUD_CLI=ibmcloud
+    export IBMCLOUD_HOME=/output
+    region="${LEASED_RESOURCE}"
+    export region
+    "${IBMCLOUD_CLI}" config --check-version=false
+    echo "Try to login..."
+    "${IBMCLOUD_CLI}" login -r ${region} --apikey @"${CLUSTER_PROFILE_DIR}/ibmcloud-api-key"
+}
+
+#OCP-60816 - [IPI-on-IBMCloud] install cluster under BYON with a different resource group
+function getResources() {
+    "${IBMCLOUD_CLI}" resource service-instances --type all --output JSON | jq -r '.[]|.name+" "+.resource_id' | sort
+}
+
+function checkResourcesInVPC() {
+    local preResourceFile resource_group curResourceFile cmpRst
+    resource_group=$1
+    preResourceFile="${SHARED_DIR}/vpc_resources"
+    "${IBMCLOUD_CLI}" target -g ${resource_group}
+    curResourceFile=$(mktemp)
+    getResources > ${curResourceFile}
+    echo "diff ${curResourceFile} ${preResourceFile} ..."
+    cmpRst=$(diff ${curResourceFile} ${preResourceFile})
+    if [ -n "$cmpRst" ]; then
+        echo "ERROR: resources differ: ${cmpRst}"
+        return 1
+    else
+        echo "checkResourcesinVPC PASS"  
+        return 0
+    fi
+}
+
+if [ -f "${SHARED_DIR}/kubeconfig" ] ; then
+    export KUBECONFIG=${SHARED_DIR}/kubeconfig
+else
+    echo "ERROR: fail to get the kubeconfig file under ${SHARED_DIR}!!"
+    exit 1
+fi
+
+if [ -f "${SHARED_DIR}/proxy-conf.sh" ] ; then
+    source "${SHARED_DIR}/proxy-conf.sh"
+fi
+
+check_result=0
+ibmcloud_login
+
+provisioned_rg_file="${SHARED_DIR}/ibmcloud_resource_group"
+cluster_resource_group=$(jq -r .ibmcloud.resourceGroupName ${SHARED_DIR}/metadata.json)
```

Comment:
I do not think we should get the value of `cluster_resource_group` from `metadata.json` when we provisioned resources group.

Reply:
${SHARED_DIR}/ibmcloud_cluster_resource_group record the cluster_resource_group
"${SHARED_DIR}/ibmcloud_resource_group" record the vpc_resource_group
when have not the above 2 files, still can check the nodes in the cluster_resource_group

Reply:
The purpose we introduced the check step is for provisioned cluster resource group, right? If right, we only take the specific case into consideration.

Reply:
if the 2 resource group files do not exist, will exit the step?

Reply:
if the 2 resource group files do not exist, the step - `cucushift-installer-check-ibmcloud-resourcegroup` is not supposed to be existing in the install workflow.

Reply:
How about still keeping this line, I will use it to check the following: 
1.  
> rgName=$(cat "${cluster_rg_file}")
    if [ "${rgName}" != "${cluster_resource_group}" ]
2. 
> if [ -n "${vpc_resource_group}" ] && [ "${vpc_resource_group}" != "${cluster_resource_group}" ];

Reply:
I already proposed the detailed codes in https://github.com/openshift/release/pull/51310/files#r1590490184, can you re-check those codes.